### DISCUSSION TITLE
Migrating Entry Types via Partial Project Config YAML Edits

### DISCUSSION THREAD

**Discussion by adrianjean** [craftcms/cms#15553] [Q&A] [answered]
Ok — I did read the [docs on this](https://craftcms.com/docs/5.x/system/project-config.html) and noticed the friendly warning that says: 

```
Never directly edit YAML files. 
Missing changes in other parts of the Project Config that should 
be made simultaneously can cause inconsistencies and instability.
```

_Ok, now hold my beer for a sec._

Let's say I had one CraftCMS 5 site (Site A) with 20 entry types that I built and configured the way I like.
Then I had another CraftCMS 5 site (Site B), built totally differently, but now I want to incorporate those entry types into my new site.

Looking at the project config folders for (Site A) — I see a folder called `entryTypes`. Inside I see my custom entry types as files, with filenames matching the field name and a unique hash:

```
cbBlockAccordion--e7d8dd0c-1a09-4e1d-9f4b-da83c1fbdc5c.yaml
```

Looking at `project.yaml` for (Site A) — I see an entry with the same hash and entry type name:

```
meta:
  __names__:
 ...
 e7d8dd0c-1a09-4e1d-9f4b-da83c1fbdc5c: Accordion # Accordion
```

**Question:**
- Can I just copy the `entryTypes/cbBlockAccordion--e7d8dd0c-1a09-4e1d-9f4b-da83c1fbdc5c.yaml` file into (Site B)'s `/entryTypes` folder, then copy the YAML entry from (Site A)'s `project.yaml` file into (Site B)'s `project.yaml` file at the end of the list of entry types, then go into `Utilities > Project Config` in the CraftCMS CP and apply the changes to "install" the new entry types into (Site B)?

**Comment by brandonkelly** [comment, marked as answer]
The `meta.__names__` values are actually unnecessary; Craft generates those values on its own.

The bigger issue is that your entry type file defines a field layout, which references custom field UUIDs (which originate from `config/project/fields/`), and your custom fields might reference other UUIDs, etc. etc..
